tests/MoBi.Tests/Presentation/Tasks/InitialConditionsTaskSpecs.cs

```diff
@@ -667,26 +668,38 @@ public void should_set_the_value_to_null()
+   public abstract class concern_for_initialConditions_With_formulas_for_Renaming : concern_for_InitialConditionsTask
    {
-      private BlackBoxFormula _replacedFormula;
-      private ExplicitFormula _refreshedFormula;
-
+      protected BlackBoxFormula _replacedFormula;
+      protected ExplicitFormula _refreshedFormula;
+      protected AddedEvent<IFormula> _addEvent;
+      protected MoleculeBuilder _builder;
       protected override void Context()
       {
          base.Context();
          var spatialStructure = A.Fake<MoBiSpatialStructure>();
          var moleculeBuildingBlock = new MoleculeBuildingBlock();
          A.CallTo(() => _context.BuildingBlockRepository.MoleculeBlockCollection).Returns(new[] { moleculeBuildingBlock });
          A.CallTo(() => _context.BuildingBlockRepository.SpatialStructureCollection).Returns(new[] { spatialStructure });
-         var builder = new MoleculeBuilder { Name = "molecule", Dimension = Constants.Dimension.NO_DIMENSION, DefaultStartFormula = new ExplicitFormula("50").WithId("50") };
-         moleculeBuildingBlock.Add(builder);
+         _builder = new MoleculeBuilder { Name = "molecule", Dimension = Constants.Dimension.NO_DIMENSION, DefaultStartFormula = new ExplicitFormula("50").WithId("50") };
+         moleculeBuildingBlock.Add(_builder);
          _replacedFormula = new BlackBoxFormula().WithId("blackbox");
-         var startValue = new InitialCondition { Name = builder.Name, Value = 45, Dimension = Constants.Dimension.NO_DIMENSION, Formula = _replacedFormula };
+         var startValue = new InitialCondition { Name = _builder.Name, Value = 45, Dimension = Constants.Dimension.NO_DIMENSION, Formula = _replacedFormula };
          _initialConditionsBuildingBlock.Add(startValue);
          _initialConditionsBuildingBlock.FormulaCache.Add(startValue.Formula);
-         _refreshedFormula = new ExplicitFormula("M/V").WithId("m/v");
-         A.CallTo(() => _cloneManagerForBuildingBlock.Clone(builder.DefaultStartFormula, _initialConditionsBuildingBlock.FormulaCache)).Invokes(x => _initialConditionsBuildingBlock.FormulaCache.Add(_refreshedFormula)).Returns(_refreshedFormula);
+         _refreshedFormula = new ExplicitFormula("M/V").WithId("m/v").WithName("FormulaName");
+      }
+
+   }
+
+   public class When_updating_a_molecule_start_value_from_original_building_block_and_all_formulas_will_be_replaced : concern_for_initialConditions_With_formulas_for_Renaming
+   {
+      protected override void Context()
+      {
+         base.Context();
+         A.CallTo(() => _cloneManagerForBuildingBlock.Clone(_builder.DefaultStartFormula, _initialConditionsBuildingBlock.FormulaCache)).Invokes(x => _initialConditionsBuildingBlock.FormulaCache.Add(_refreshedFormula)).Returns(_refreshedFormula);
+         A.CallTo(() => _context.Context.PublishEvent(A<AddedEvent<IFormula>>._))
+            .Invokes(x => _addEvent = x.GetArgument<AddedEvent<IFormula>>(0));
```

Comment:
Could you move `.Invokes` to the second line to be consistent with the second `A.CallTo` because I think they both benefit from the additional line due to length.